test/root/RootChain.test.js

```diff
@@ -62,24 +64,43 @@ contract('RootChain', async function (accounts) {
     }
   })
 
+  it('try to submitHeaderBlock with negative vote(`00` prefix) and fail', async function () {
+    const validators = [1, 2, 3, 4]
+    let tree = await rewradsTree(validators, accountState)
+    const { data, sigs } = buildSubmitHeaderBlockPaylod(
+      accounts[0],
+      0,
+      255,
+      '' /* root */,
+      wallets,
+      { allValidators: true, rewardsRootHash: tree.getRoot(), getSigs: true, totalStake: totalStake, sigPrefix: '0x00' }
+    )
+    try {
+      await rootChain.submitHeaderBlock(data, sigs)
+    } catch (error) {
+      const invalidOpcode = error.message.search('revert') >= 0
+      assert(invalidOpcode, "Expected revert, got '" + error + "' instead")
+    }
+  })
+
   it('submitHeaderBlock', async function () {
     const validators = [1, 2, 3, 4]
     let tree = await rewradsTree(validators, accountState)
-    const { vote, sigs, extraData, root } = buildSubmitHeaderBlockPaylod(
+    const { data, sigs } = buildSubmitHeaderBlockPaylod(
       accounts[0],
       0,
       255,
       '' /* root */,
       wallets,
       { allValidators: true, rewardsRootHash: tree.getRoot(), getSigs: true, totalStake: totalStake }
     )
-    const result = await rootChain.submitHeaderBlock(vote, sigs, extraData)
+    const result = await rootChain.submitHeaderBlock(data, sigs)
     const logs = result.logs
     logs.should.have.lengthOf(1)
     logs[0].event.should.equal('NewHeaderBlock')
     expect(logs[0].args).to.include({
       proposer: accounts[0],
-      root: '0x' + root.toString('hex')
+      root: ethUtils.bufferToHex(ethUtils.keccak256(encode(0, 255)))
```

Comment:
This is just a dummy root assertion right? can we remove it?

Reply:
why remove? this is correct only.